samples/bluetooth/hci_spi/src/main.c

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <stddef.h>
+#include <stdio.h>
+
+#include <zephyr.h>
+#include <misc/byteorder.h>
+#include <logging/sys_log.h>
+#include <misc/stack.h>
+
+#include <device.h>
+#include <init.h>
+#include <gpio.h>
+#include <spi.h>
+
+#include <net/buf.h>
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/l2cap.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/buf.h>
+#include <bluetooth/hci_raw.h>
+#include <bluetooth/common/log.h>
+
+#define HCI_CMD			0x01
+#define HCI_ACL			0x02
+#define HCI_SCO			0x03
+#define HCI_EVT			0x04
+
+/* Special Values */
+#define SPI_WRITE		0x0A
+#define SPI_READ		0x0B
+#define READY_NOW		0x02
+#define SANITY_CHECK		0x02
+
+/* Offsets */
+#define STATUS_HEADER_READY	0
+#define STATUS_HEADER_TOREAD	3
+
+#define PACKET_TYPE		0
+#define EVT_BLUE_INITIALIZED    0x01
+
+#define GPIO_IRQ_DEV_NAME	CONFIG_BLUETOOTH_SPI_TO_HOST_IRQ_DEV_NAME
+#define GPIO_IRQ_PIN		CONFIG_BLUETOOTH_SPI_TO_HOST_IRQ_PIN
+
+/* Needs to be aligned with the SPI master buffer size */
+#define SPI_MAX_MSG_LEN		255
+
+static u8_t rxmsg[SPI_MAX_MSG_LEN];
+static u8_t txmsg[SPI_MAX_MSG_LEN];
+
+/* HCI buffer pools */
+#define CMD_BUF_SIZE BT_BUF_RX_SIZE
+
+NET_BUF_POOL_DEFINE(cmd_tx_pool, CONFIG_BLUETOOTH_HCI_CMD_COUNT, CMD_BUF_SIZE,
+		    BT_BUF_USER_DATA_MIN, NULL);
+
+#define BT_L2CAP_MTU 65 /* 64-byte public key + opcode */
```

Comment:
There is a conditional compilation introduced in hci_uart to pick the MTU from controller's configured tx buffer size, please use that code from hci_uart